### ISSUE TITLE
chunk option `validate=TRUE` always shows validations passed for agents, even when they don't

### ISSUE THREAD

**Issue by Aariq** [rstudio/pointblank#370]
## Prework

* [x] Read and agree to the [code of conduct](https://www.contributor-covenant.org/version/2/0/code_of_conduct/) and [contributing guidelines](https://github.com/rich-iannone/pointblank/blob/master/.github/CONTRIBUTING.md).
* [ ] If there is [already a relevant issue](https://github.com/rich-iannone/pointblank/issues), whether open or closed, comment on the existing thread instead of posting a new issue.
* [x] Post a [minimal reproducible example](https://www.tidyverse.org/help/) so the maintainer can troubleshoot the problems you identify. A reproducible example is:
    * [x] **Runnable**: post enough R code and data so any onlooker can create the error on their own computer.
    * [x] **Minimal**: reduce runtime wherever possible and remove complicated details that are irrelevant to the issue at hand.
    * [x] **Readable**: format your code according to the [tidyverse style guide](https://style.tidyverse.org/).

## Description

The RMarkdown template in `pointblank()` has an example where an agent is used in a validate chunk.  If that agent has validations steps that fail, the rendered markdown doc still  says "All validations passed."

## Reproducible example
````
```{r setup, include=FALSE}
library(pointblank)
```

```{r validate=TRUE}
small_table %>%
  create_agent(actions = action_levels(stop_at = 1)) %>%
  col_vals_gt(vars(a), 30000) %>% interrogate()
```
````

Knitting this shows "all validations passed"

## Expected result

Running just the validation chunk in-line shows that the one validation step fails, I would expect the knitted document to reflect that.

## Session info

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
```

    #> ─ Session info ───────────────────────────────────────────────────────────────
    #>  setting  value                       
    #>  version  R version 4.0.2 (2020-06-22)
    #>  os       macOS  10.16                
    #>  system   x86_64, darwin17.0          
    #>  ui       X11                         
    #>  language (EN)                        
    #>  collate  en_US.UTF-8                 
    #>  ctype    en_US.UTF-8                 
    #>  tz       America/New_York            
    #>  date     2021-11-29                  
    #> 
    #> ─ Packages ───────────────────────────────────────────────────────────────────
    #>  package     * version    date       lib
    #>  assertthat    0.2.1      2019-03-21 [1]
    #>  backports     1.4.0      2021-11-23 [1]
    #>  base64enc     0.1-3      2015-07-28 [1]
    #>  blastula      0.3.2      2020-05-19 [1]
    #>  checkmate     2.0.0      2020-02-06 [1]
    #>  cli           3.1.0      2021-10-27 [1]
    #>  colorspace    2.0-2      2021-06-24 [1]
    #>  commonmark    1.7        2018-12-01 [1]
    #>  crayon        1.4.2      2021-10-29 [1]
    #>  DBI           1.1.1      2021-01-15 [1]
    #>  digest        0.6.28     2021-09-23 [1]
    #>  dplyr         1.0.7      2021-06-18 [1]
    #>  ellipsis      0.3.2      2021-04-29 [1]
    #>  evaluate      0.14       2019-05-28 [1]
    #>  fansi         0.5.0      2021-05-25 [1]
    #>  fastmap       1.1.0      2021-01-25 [1]
    #>  fs            1.5.0      2020-07-31 [1]
    #>  generics      0.1.1      2021-10-25 [1]
    #>  ggplot2       3.3.5      2021-06-25 [1]
    #>  glue          1.5.0      2021-11-07 [1]
    #>  gt            0.3.1      2021-08-07 [1]
    #>  gtable        0.3.0      2019-03-25 [1]
    #>  highr         0.9        2021-04-16 [1]
    #>  htmltools     0.5.2      2021-08-25 [1]
    #>  knitr         1.36       2021-09-29 [1]
    #>  lifecycle     1.0.1      2021-09-24 [1]
    #>  magrittr      2.0.1      2020-11-17 [1]
    #>  munsell       0.5.0      2018-06-12 [1]
    #>  pillar        1.6.4      2021-10-18 [1]
    #>  pkgconfig     2.0.3      2019-09-22 [1]
    #>  pointblank  * 0.9.0.9000 2021-11-29 [1]
    #>  purrr         0.3.4      2020-04-17 [1]
    #>  R6            2.5.1      2021-08-19 [1]
    #>  reprex        2.0.1      2021-08-05 [1]
    #>  rlang         0.4.12     2021-10-18 [1]
    #>  rmarkdown     2.11       2021-09-14 [1]
    #>  rstudioapi    0.13       2020-11-12 [1]
    #>  sass          0.4.0      2021-05-12 [1]
    #>  scales        1.1.1      2020-05-11 [1]
    #>  sessioninfo   1.1.1      2018-11-05 [1]
    #>  stringi       1.7.5      2021-10-04 [1]
    #>  stringr       1.4.0      2019-02-10 [1]
    #>  styler        1.5.1      2021-07-13 [1]
    #>  tibble        3.1.6      2021-11-07 [1]
    #>  tidyselect    1.1.1      2021-04-30 [1]
    #>  utf8          1.2.2      2021-07-24 [1]
    #>  vctrs         0.3.8      2021-04-29 [1]
    #>  withr         2.4.2      2021-04-18 [1]
    #>  xfun          0.28       2021-11-04 [1]
    #>  yaml          2.2.1      2020-02-01 [1]
    #>  source                                  
    #>  CRAN (R 4.0.0)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.0)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.0)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.0)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.0)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.0)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.0)                          
    #>  Github (rich-iannone/pointblank@97d4bf5)
    #>  CRAN (R 4.0.0)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.0)                          
    #>  CRAN (R 4.0.0)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.0)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.2)                          
    #>  CRAN (R 4.0.0)                          
    #> 
    #> [1] /Library/Frameworks/R.framework/Versions/4.0/Resources/library

</details>

**Comment by rich-iannone** [comment]
With the RMarkdown template, the message that gathers all passes and failures only looks at validations done directly at the data. So if the chunk had:


````
```{r validate=TRUE}
small_table %>%
  col_vals_gt(vars(a), 30000)
```
````

then you'd see that the failure would be captured. So agent-based workflows are ignored. However, this is weird and unexpected. Do you think I should provide a summary of how many agents are created and perhaps some more detailed info in this message?

**Comment by Aariq** [comment]
> So agent-based workflows are ignored.

I guess I didn't understand that from the template which says: "We can provide a single, agent-based interrogation inside of a `validate` chunk. Don't put anything else in the chunk though (the agent likes to work alone)."   I expected agent-based workflows to work in `validate=TRUE` chunks just like the other examples.

**Comment by rich-iannone** [comment]
Yeah, the template is not really upfront on what's going on with agents. I'll see if I can generally improve the summary and documentation a bit (alongside the fixes required for https://github.com/rich-iannone/pointblank/issues/369).

**Comment by rich-iannone** [comment]
I figured out what was going on (and fixed things)! Turns out the detection of an agent report became broken so that it always said a chunk with it would pass. I fixed that and now it just says `agent report`.

I also adjusted some of the language in the template concerning agent reports and their behavior in validate chunks.

Also, we now need to use `validate_rmd()` in the .Rmd to enable parsing of validate code chunks (I’ve included this in the revised template).
